src/Databases/DatabaseAtomic.cpp

```diff
@@ -483,33 +496,41 @@ UUID DatabaseAtomic::tryGetTableUUID(const String & table_name) const
 
 void DatabaseAtomic::beforeLoadingMetadata(ContextMutablePtr /*context*/, LoadingStrictnessLevel mode)
 {
+    auto shared_disk = getContext()->getSharedDisk();
     if (mode < LoadingStrictnessLevel::FORCE_RESTORE)
         return;
 
-    if (!fs::exists(path_to_table_symlinks))
+    if (!shared_disk->isSymlinkSupported())
+        return;
+
+    if (!shared_disk->existsFileOrDirectory(path_to_table_symlinks))
         return;
 
     /// Recreate symlinks to table data dirs in case of force restore, because some of them may be broken
-    for (const auto & table_path : fs::directory_iterator(path_to_table_symlinks))
+    for (const auto dir_it = shared_disk->iterateDirectory(path_to_table_symlinks); dir_it->isValid(); dir_it->next())
     {
-        if (!FS::isSymlink(table_path))
+        auto table_path = fs::path(dir_it->path());
+        if (table_path.filename().empty())
+            table_path = table_path.parent_path();
+        if (!shared_disk->isSymlink(table_path))
         {
-            throw Exception(ErrorCodes::ABORTED,
-                "'{}' is not a symlink. Atomic database should contains only symlinks.", std::string(table_path.path()));
+            throw Exception(
+                ErrorCodes::ABORTED, "'{}' is not a symlink. Atomic database should contains only symlinks.", std::string(table_path));
         }
 
-        (void)fs::remove(table_path);
+        (void)shared_disk->removeFileIfExists(table_path);
```

Comment:
```suggestion
        (void)shared_disk->removeFile(table_path);
```

Reply:
In `DiskLocal` which is a subclass of `IDisk`, we use `unlink` to `removeFile`. If the file does not exist, it returns an error, and we will thrown an exception.
However, in `fs::remove`, it will return `false` only.
So I use `removeFileIfExists` to make the behavior unchanged.